params/config.go

```diff
@@ -865,3 +1018,36 @@ func (c *ChainConfig) ToWithUpgradesJSON() *ChainConfigWithUpgradesJSON {
 		UpgradeConfig: c.UpgradeConfig,
 	}
 }
+
+// Checks if messages have the same hash
+//
+// `message` is the simulation of a configuration being parsed from the local
+// config. `message2` is parsing a message being exchanged through the network
+// (a foreign config), and `message3` is the the deserialization and
+// serialization of the foreign config. All 3 instances should have the same
+// hashing, depite maybe not being identical (some configurations may be in a
+// different order, but our hashing algorithm is resilient to those changes,
+// thanks for our serialization library, which produces always the same output.
```

Comment:
Could you re-phrase this comment, it seems like some variable names have changed

Reply:
I think I made it way better in 1819327a